PIL/ImageQt.py

```diff
@@ -49,54 +134,30 @@ def rgb(r, g, b, a=255):
 # @param im A PIL Image object, or a file name (given either as Python
 #     string or a PyQt string object).
 
-class ImageQt(QImage):
-
-    def __init__(self, im):
-
-        data = None
-        colortable = None
-
-        # handle filename, if given instead of image name
-        if hasattr(im, "toUtf8"):
-            # FIXME - is this really the best way to do this?
-            if str is bytes:
-                im = unicode(im.toUtf8(), "utf-8")
-            else:
-                im = str(im.toUtf8(), "utf-8")
-        if isPath(im):
-            im = Image.open(im)
-
-        if im.mode == "1":
-            format = QImage.Format_Mono
-        elif im.mode == "L":
-            format = QImage.Format_Indexed8
-            colortable = []
-            for i in range(256):
-                colortable.append(rgb(i, i, i))
-        elif im.mode == "P":
-            format = QImage.Format_Indexed8
-            colortable = []
-            palette = im.getpalette()
-            for i in range(0, len(palette), 3):
-                colortable.append(rgb(*palette[i:i+3]))
-        elif im.mode == "RGB":
-            data = im.tobytes("raw", "BGRX")
-            format = QImage.Format_RGB32
-        elif im.mode == "RGBA":
-            try:
-                data = im.tobytes("raw", "BGRA")
-            except SystemError:
-                # workaround for earlier versions
-                r, g, b, a = im.split()
-                im = Image.merge("RGBA", (b, g, r, a))
-            format = QImage.Format_ARGB32
-        else:
-            raise ValueError("unsupported image mode %r" % im.mode)
-
-        # must keep a reference, or Qt will crash!
-        self.__data = data or im.tobytes()
-
-        QImage.__init__(self, self.__data, im.size[0], im.size[1], format)
-
-        if colortable:
-            self.setColorTable(colortable)
+if qt_is_installed:
+    class ImageQt(QImage):
+
+        def __init__(self, im):
+            im_data = _toqclass_helper(im)
+            QImage.__init__(self,
+                            im_data['data'], im_data['im'].size[0],
+                            im_data['im'].size[1], im_data['format'])
+            if im_data['colortable']:
+                self.setColorTable(im_data['colortable'])
+
+
+def toqimage(im):
+    return ImageQt(im)
+
+
+def toqpixmap(im):
+    #   This doesn't work. For now using a dumb approach.
+    # im_data = _toqclass_helper(im)
+    # result = QPixmap(im_data['im'].size[0], im_data['im'].size[1])
+    # result.loadFromData(im_data['data'])
+    # Fix some strange bug that causes
+    if im.mode == 'RGB':
+        im = im.convert('RGBA')
+    qimage = toqimage(im)
+    qimage.save('/tmp/hopper_{}_qpixmap_qimage.png'.format(im.mode))
```

Comment:
I don't even think this line is necessary. I'd been working on simply finishing the original code of this PR, repairing rather than rewriting, so thanks for picking this up. I've taken a further look at the original code, and hopefully made some improvements.